nixos/modules/hardware/tuxedo-keyboard.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+# TODO: look at the other hardware modules
+# TODO: re-build for the specific kernel
+# TODO: I am not quite sure IF a module is still needed here since the configurations here could be set directly without this module. 
+# 
+#         boot.kernelParams = [
+#           "tuxedo_keyboard.mode=0"
+#           "tuxedo_keyboard.brightness=${toString cfg.brightness}"
+#           "tuxedo_keyboard.color_left=0x${cfg.color}"
+#         ]; 
+with lib;
+
+let 
+  # TODO: load tuxedo-keyboard from the kernel Packages
```

Comment:
that's solved by loading the tuxedo-keyboard from the `config.boot.kernelPackages` option.
```suggestion
```